internal/api/ratelimits.go

```diff
@@ -0,0 +1,46 @@
+package api
+
+import (
+	"sync/atomic"
+	"time"
+
+	"github.com/supabase/auth/internal/conf"
+)
+
+type SimpleRateLimiter struct {
+	Rate conf.Rate
+
+	counter uint64
+}
+
+func NewSimpleRateLimiter(now time.Time, rate conf.Rate) *SimpleRateLimiter {
+	r := &SimpleRateLimiter{
+		Rate: rate,
+	}
+
+	counterStartedAt := now.Truncate(rate.OverTime)
+	counterResetsAt := counterStartedAt.Add(rate.OverTime)
+
+	proRate := float64(counterStartedAt.Sub(now).Milliseconds()) / float64(rate.OverTime.Milliseconds())
+
+	r.counter = uint64(rate.Events * proRate)
+
+	go r.fillBucket(counterResetsAt.Sub(now))
+
+	return nil
```

Comment:
should this return the rate limiter?
```suggestion
	return r
```